pkg/kapp/app/recorded_app.go

```diff
@@ -144,14 +143,37 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string, isDiffRun bool) (
 	if err != nil {
 		return false, err
 	}
-
 	if foundNonMigratedApp {
 		if a.isMigrationEnabled() {
 			return false, a.migrate(app, labels, a.fqName())
 		}
 		return false, a.updateApp(app, labels)
 	}
 
+	if prevAppName == "" {
+		return true, a.create(labels, isDiffRun)
```

Comment:
Should we move all the logic in lines 157-176 inside this block? 
Taking a closer look but it feels that we are always returning here if prevAppName is not empty. The logic which follows after would be unreachable and is only valid if `prevAppName` is not empty?

Reply:
> The logic which follows after would be unreachable and is only valid if prevAppName is not empty?

Yep, that's the intention.

Reply:
If we always return when `prevApp` name is not empty. When are the lines below ever helpful?

Reply:
> If we always return when prevApp name is not empty

We are returning when `prevAppName` _is_ empty 🤔